botocore/credentials.py

```diff
@@ -791,6 +820,206 @@ def _assume_role_base_kwargs(self, config):
         return assume_role_kwargs
 
 
+def _role_selector(role_arn, roles):
```

Comment:
pep8, functions at the top, not inbetween classes.

also docstring conventions (pep257) summary on one line, blank line, remaining docs.